url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/net/xeoh/plugins/diagnosis/local/impl/serialization/java/LogFileWriter.java

```diff
@@ -0,0 +1,131 @@
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+package net.xeoh.plugins.diagnosis.local.impl.serialization.java;
+
+//import static net.jcores.jre.CoreKeeper.$;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.zip.GZIPOutputStream;
+
+public class LogFileWriter {
+
+    /** Raw file to write into */
+	private FileOutputStream fileOutputStream;
+
+    /** Object stream to use when writing */
+	private ObjectOutputStream objectOutputStream;
+
+    /** Zip stream to compress */
+	private GZIPOutputStream zipStream;
+
+    /** Entries to write into the file */
+	private LinkedBlockingQueue<Entry> eventQueue = new LinkedBlockingQueue<Entry>();
+
+    /**
+     * Creates a new serializer
+     * 
+     * @param file The file to write into.
+     * @param compressOutput
+     */
+    public LogFileWriter(String file, boolean compressOutput) {
+        try {
+            //this.fileOutputStream = new FileOutputStream($(file).get("diagnosis.fallback.record"));
+        	this.fileOutputStream = new FileOutputStream(file);
+            
+            if (compressOutput) {
+                this.zipStream = new GZIPOutputStream(this.fileOutputStream);
+                this.objectOutputStream = new ObjectOutputStream(this.zipStream);
+            } else
+                this.objectOutputStream = new ObjectOutputStream(this.fileOutputStream);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        final Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                int flushCount = 0;
+                while (true) {
+                    try {
+                        final Entry take = LogFileWriter.this.eventQueue.take();
+                        LogFileWriter.this.objectOutputStream.writeUnshared(take);
+
+                        if (flushCount++ > 50) {
+                            LogFileWriter.this.objectOutputStream.reset();
+                            LogFileWriter.this.objectOutputStream.flush();
+                            if (LogFileWriter.this.zipStream != null) {
+                                LogFileWriter.this.zipStream.flush();
+                            }
+                            LogFileWriter.this.fileOutputStream.flush();
+                            flushCount = 0;
+                        }
+                    }
+                    catch (InterruptedException e) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)